content/pytorch/concepts/datasets-and-dataloaders/datasets-and-dataloaders.md

```diff
@@ -0,0 +1,83 @@
+---
+Title: 'PyTorch Datasets and DataLoaders'
+Description: 'An overview of PyTorch Datasets and DataLoaders, including how to create custom datasets and use DataLoader for efficient data handling.'
+Subjects: 
+  - 'Data Science'
+  - 'Machine Learning'
+Tags: 
+  - 'PyTorch'
+  - 'Datasets'
+  - 'DataLoader'
+CatalogContent: 
+  - 'learn-pytorch'
+  - 'paths/deep-learning'
+---
+
+**PyTorch** provides two essential data handling abstractions: `torch.utils.data.Dataset` and `torch.utils.data.DataLoader`. These tools are crucial for efficient data management and preprocessing during model training.
+
+## Creating a Custom Dataset
+
+To create a custom dataset in PyTorch, subclass `torch.utils.data.Dataset` and override the following methods:
+
+- `__init__(self)`: Initializes the dataset, typically loading data into memory or setting up file paths.
+- `__len__(self)`: Returns the total number of samples in the dataset.
+- `__getitem__(self, idx)`: Retrieves a sample and its corresponding label at the specified index `idx`.
+
+### Example: Custom Dataset Class
+
+```python
+import torch
+from torch.utils.data import Dataset
+
+class CustomDataset(Dataset):
+    def __init__(self, data, labels):
+        self.data = data
+        self.labels = labels
+
+    def __len__(self):
+        return len(self.data)
+
+    def __getitem__(self, idx):
+        sample = self.data[idx]
+        label = self.labels[idx]
+        return sample, label
+```

```

Comment:
``` suggestion
```py
import torch
from torch. utils.data import Dataset

class CustomDataset(Dataset):
    def __init__(self, data, labels):
        self.data = data
        self.labels = labels

    def __len__(self):
        return len(self.data)

    def __getitem__(self, IDX):
        sample = self.data[IDX]
        label = self.labels[IDX]
        return sample, label
```

```
The supported code block for `python` is ```py```, so the code should be written inside ``` code block instead of writing it inside ```python``` code block.

In the above code, there is a small typo mistake, the parameter `IDX` is defined in the `getitem` method, but when you're referencing it inside the method you are referring to it as `idx` which is not correct. We should use the same variable name inside the method that we have used at the time of the variable declaration.